api/spec/controllers/mno_enterprise/jpi/v2/users_controller_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+module MnoEnterprise
+  describe Jpi::V2::UsersController, type: :controller do
+    include MnoEnterprise::TestingSupport::SharedExamples::JpiV2ApiController
+
+    it_behaves_like MnoEnterprise::Jpi::V2::ApiController, %i[index show update]
+
+    include_context 'v2 api controller requests context'
+
+    describe 'GET #show' do
+      subject { get :show, id: record.id }
+
+      let(:stub_resp) do
+        { 'data' => { 'attributes' => { 'name' => user.name } } }
+      end
+
+      let(:stub) do
+        stub_request(:get, File.join(base_url, endpoint, record.id))
+          .with(headers: headers)
+          .to_return(status: 200, body: stub_resp.to_json, headers: {})
+      end
+
+      it { is_expected.to have_http_status(:ok) }
+      it { expect(stub).to have_been_requested }
+      it { expect(JSON.parse(response.body)).to eq(stub_resp) }
+
+      context 'when intercom is enabled' do
+        before do
+          allow_any_instance_of(MnoEnterprise::IntercomUserIdentity).to receive(:new).and_return(
```

Comment:
Ah it's because you mixing `allow_any_instance_of` with `new`

So either:
```ruby
allow(MnoEnterprise::IntercomUserIdentity).to receive(:new).and_return(double(....

# OR

allow_any_instance_of(MnoEnterprise::IntercomUserIdentity).to receive(:intercom_enabled?).and_return(true)

```